chain-factory/src/factory.rs

```diff
@@ -1,14 +1,22 @@
 use chain_config::StakeMultiArg;
 
+use multiversx_sc::imports::*;
+use multiversx_sc_modules::only_admin;
 multiversx_sc::derive_imports!();
-multiversx_sc::imports!();
 
 #[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, ManagedVecItem)]
-struct ContractMapArgs<M: ManagedTypeApi> {
+pub struct ContractMapArgs<M: ManagedTypeApi> {
     id: ScArray,
     address: ManagedAddress<M>,
 }
 
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, ManagedVecItem)]
```

Comment:
I think TypeAbi should be remove from here and add as `#[type_abi]`

Reply:
done